### PR TITLE
feat(jdbc): setDate

### PR DESCRIPTION
**Pull request by Mause**
Resolves https://github.com/duckdb/duckdb/discussions/7207

### CODE REVIEW

File: tools/jdbc/src/jni/duckdb_java.cpp [hunk 1]
```diff
@@ -83,11 +83,27 @@ static jclass J_UUID;
...
+static jmethodID J_DuckDBDate_getDaysSinceEpoch;
+
 void ThrowJNI(JNIEnv *env, const char *message) {
 	D_ASSERT(J_SQLException);
 	env->ThrowNew(J_SQLException, message);
 }
 
+static duckdb::vector<jclass> toFree;
+
+static jclass GetClassRef(JNIEnv *env, const string &name) {
```
Comment (Mause) [inline tools/jdbc/src/jni/duckdb_java.cpp:96]: Thoughts on this @Mytherin? Right now we're not really cleaning up any of the global references, and I figure something like this makes it easier, plus it removes some boilerplate

Comment (Mytherin) [reply tools/jdbc/src/jni/duckdb_java.cpp:96]: I have no concept of what a global reference in Java is, but I can guess :)

What I would suggest is we should create a `GlobalReferenceManager` singleton class that holds the references and correctly grabs locks. Perhaps they can even be cached similar to how the Python import caches work?

**Comment by Mytherin** [comment]
@Mause Would you like this to be merged as-is or would you like to work on a different solution for the global references?

**Comment by Mause** [comment]
> @Mause Would you like this to be merged as-is or would you like to work on a different solution for the global references?

I think merging as is would be good, I'm investigating jnipp to improve things in the future

**Comment by Mytherin** [comment]
Alright, thanks!

**Pull request merged**